### ISSUE TITLE
Bit-depth is not detected correctly by pydub

### ISSUE THREAD

**Issue by aspen1135** [jiaaro/pydub#381]
### Steps to reproduce
`
def open_audio(file_path):

    audio = pd.AudioSegment.from_file(file_path)
    return audio

def get_bit_depth(file):

    bit_depth = file.sample_width  # 1-4. 1=8 2=16 3=24 4=32

    if bit_depth is 1:
        bit_depth = 8

    elif bit_depth is 2:
        bit_depth = 16

    elif bit_depth is 3:
        bit_depth = 24

    elif bit_depth is 4:
        bit_depth = 32

    return bit_depth

file = pd.open_audio("PUT FILE PATH HERE")
bit_depth = pd.get_bit_depth(file)

print(bit_depth)
`

### Expected behavior
Should print the actual bit-depth of the file (not the 1-4 value).

### Actual behavior
I've tested 4 different files with 8 bit, 16 bit, 24 bit, and 32 bit depths.

8 bit is detected correctly, but whenever exported the audio becomes garbled and in-comprehensive compared to the source file. It should stay 1:1, especially if assigned higher bit-depth.
16 bit is detected correctly
24 bit is detected as 32 bit file (exported using ableton DAW, also tested with third-party converter)
32 bit file is detected correctly.

### Your System configuration
- Python version: 3.7
- Pydub version: 0.23.1
- ffmpeg or avlib?: ffmpeg
- ffmpeg/avlib version: 4.1.3 win64 static (stable distro, not nightly)

### Is there an audio file you can include to help us reproduce?
4 different samples, one of every bit-depth type.
[Test sample (kick drum).zip](https://github.com/jiaaro/pydub/files/3144955/Test.sample.kick.drum.zip)

**Comment by jiaaro** [comment]
Hi, thanks for writing :)

Pydub uses Python's stdlib `audioop` module for most operations, and unfortunately audioop does not have support for 24-bit audio (at least, in many versions of python, CPython added support for 24-bit audio in Python 3.4). So the decision was made to convert 24 bit audio to 32 bit to avoid losing quality. 

It may be time to revisit the decision perhaps in favor of an approach that uses 24-bit audio in environments that support it. 

We could check for 24-bit support relatively easily I think:

```python
import audioop

def supports_24_bit_audio():
    try:
        audioop.add(b'', b'', 3)
        return True
    except audioop.error:
        return False
```

**Comment by aspen1135** [comment]
Yeah that is a little disappointing. I may have to use a different module to handle 24 bit files then, hopefully you guys can update it with a workaround.

Any reason why the 8 bit files are exporting out garbled? I'm not sure if the kick drum is a long enough example to really reveal what I mean.

**Comment by aspen1135** [comment]
Disreguard my last msg. I haven't had time to try this and see if it will then detect 24 bit audio.

Is audioop also used for the export function? Will I also have to tweak that too?

Thanks for the reply

**Comment by jiaaro** [comment]
it's difficult to tell from the kick drum clips, but my best guess is that the 8-bit clip sounds bad simply because of it's (very) small dynamic range, 16 bit really is the lowest reasonable bit depth for good quality audio.

**Comment by jiaaro** [comment]
audioop is not used during export but pydub currently converts 24-bit audio data to 32 bit when you instantiate the AudioSegment in order to ensure all audio operation will work properly

**Comment by aspen1135** [comment]
@jiaaro

It's been a few months since I made this thread. I wanted to see if an update has been added to support 24 bit conversion.

I'm using another library right now called PyAV which gives me some python bindings to FFmpeg's framekwork. And although it supports 24 bit (only when using pcm_s24le encoder), it still has limited supported for other features I'd like to implement into my project.

Has there been any update on this?
